src/plotman/archive.py

```diff
@@ -87,40 +87,41 @@ def compute_priority(phase, gb_free, n_plots):
 
 def get_archdir_freebytes(arch_cfg):
     archdir_freebytes = {}
-    df_cmd = ('ssh %s@%s df -aBK | grep " %s/"' %
-        (arch_cfg.rsyncd_user, arch_cfg.rsyncd_host, posixpath.normpath(arch_cfg.rsyncd_path)) )
-    with subprocess.Popen(df_cmd, shell=True, stdout=subprocess.PIPE) as proc:
-        for line in proc.stdout.readlines():
-            fields = line.split()
-            if fields[3] == b'-':
-                # not actually mounted
-                continue
-            freebytes = int(fields[3][:-1]) * 1024  # Strip the final 'K'
-            archdir = (fields[5]).decode('utf-8')
-            archdir_freebytes[archdir] = freebytes
-    return archdir_freebytes
+    completed_process = subprocess.run(
+        [arch_cfg.disk_space_path],
+        encoding='utf-8',
+        env={**os.environ, **arch_cfg.environment()},
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+    )
+
+    for line in completed_process.stdout.splitlines():
+        split = line.split(':')
+        if len(split) != 2:
+            # TODO: warning?  or something...
```

Comment:
Would it make sense to test all the archive configurations during startup of one small hidden file (e.g., `.plotman-archive-test`)?

This could be triggered before the main plots command or the archive command at least to ensure archiving will work and can help folks quickly test their config. Alternatively, a dedicated archive-test command would print more verbose errors regarding the archiving configuration.

Reply:
I had considered a `plotman config check` command or such as a separate activity.  Doing an automatic precheck on startup for some things may also be worthwhile, perhaps, but I think I would leave that for a separate effort.

Reply:
(where `check` would involve doing stuff like writing a fake log (and removing it) and so on like you suggested for archiving, not just static checks of the config)